plugins/default/lib/component_health.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+import sys
+import json
+import argparse
+import urllib2
+
+import nagios
+
+class RequestError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-H","--host", action="store", required=True, help="Host")
+    parser.add_argument('-P','--port', action="store", required=True, help="Port")
+    parser.add_argument("-E","--endpoint", action="store", required=False, help="Endpoint", default="/sys/info/health")
+    return parser
+
+def do_request(host, port, endpoint):
+    url =  'http://%s:%s%s' % (host,port,endpoint)
+    try:
+        data = urllib2.urlopen(url).read()
+    except urllib2.HTTPError, error:
+        if error.code == 500:
+            # For some reason the health endpoint returns a http status code of 500 when the health isn't ok, so we have to check the http error for a health response here
+            try:
+                data = error.read()
```

Comment:
The health endpoint returns a http status of 500 if the health is deemed to be anything other than 'ok', even though the http request is perfectly fine and there is no internal server error at the application level. It seems to be a design thing to use the status codes in this way. I wouldn't agree with it, just makes things harder,  but that's the way it is.

The urllib2 error contains the json response (health response) in the error see http://stackoverflow.com/questions/2233687/overriding-urllib2-httperror-and-reading-response-html-anyway